// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.scp.core.journal_service;

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

message LastCheckpointMetadata {
  uint64 last_checkpoint_id = 1;
}

message CheckpointMetadata {
  uint64 last_processed_journal_id = 1;
}

message JournalLog {
  uint64 type = 1;
  bytes log_body = 2;
}
